ui/src/views/image/AddKubernetesSupportedVersion.vue

```diff
@@ -207,23 +218,30 @@ export default {
         }
       })
     },
+    handleZoneChange (zoneIdx) {
+      const zone = this.zones[zoneIdx]
+      if (zone.type && zone.type === 'Edge') {
+        this.form.directdownload = true
+        this.directDownloadDisabled = true
+        return
+      }
+      this.form.directdownload = false
+      this.directDownloadDisabled = false
+    },
```

Comment:
The value directDownload is set to false , when the prior selection was an edge zone and the current is not. But what if before it was already selected to be true, then edge zone was selected and deselected again?

Reply:
[lastest](https://github.com/apache/cloudstack/pull/8142/commits/af37049643b26669f5cb4e6e574cbe3296e06f51) commit should take care of this